### PR TITLE
[test] Add test for Catmem LibOS cleanup

### PR DESCRIPTION
**Pull request by iyzhang**
This PR adds a test for intentionally leaving a pipe open after the LibOS goes out of scope. We use the same name to try to open the pipe again. Assuming that the LibOS correctly cleans up resources on exit, this test should be able to create another pipe with the same name.

### CODE REVIEW

File: tests/rust/pipe-test/create_pipe/mod.rs [hunk 1]
```diff
@@ -50,15 +66,35 @@ fn create_pipe_with_same_name(libos: &mut LibOS, pipe_name: &str) -> Result<()>
...
+    Ok(())
+}
+
+/// Helper function that creates a pipe.
+fn create_pipe(libos: &mut LibOS, pipe_name: &str) -> Result<QDesc> {
+    match libos.create_pipe(pipe_name) {
+        Ok(pipeqd) => Ok(pipeqd),
+        Err(e) => anyhow::bail!("create_pipe() failed ({})", e),
+    }
+}
```
Comment (ppenna) [inline tests/rust/pipe-test/create_pipe/mod.rs:100]: Couldn't we have just two functions?
- `create_pipe()`, which creates a pipe.
- `create_pipe_and_close()`, which creates and closes a pipe.

File: tests/rust/pipe-test/create_pipe/mod.rs [hunk 2]
```diff
@@ -19,6 +20,24 @@ use ::demikernel::{
 pub fn run(libos: &mut LibOS, pipe_name: &str) -> Vec<(String, String, Result<(), anyhow::Error>)> {
     let mut result: Vec<(String, String, Result<(), anyhow::Error>)> = Vec::new();
 
+    {
+        let libos: &mut LibOS = &mut {
+            // Ok to use expect here because we should have parsed the LibOSName previously
```
Comment (ppenna) [inline tests/rust/pipe-test/create_pipe/mod.rs:25]: Would you add a full stop at the end of the line?

File: tests/rust/pipe-test/create_pipe/mod.rs [hunk 3]
```diff
@@ -19,6 +20,24 @@ use ::demikernel::{
...
+            let libos_name: LibOSName = LibOSName::from_env().expect("Should have a valid LibOS name").into();
+            LibOS::new(libos_name).expect("Should be able to create another libOS")
+        };
+
+        crate::collect!(result, crate::test!(create_pipe_with_name(libos, pipe_name)));
+    }
+    {
+        let libos: &mut LibOS = &mut {
+            // Ok to use expect here because we should have parsed the LibOSName previously
```
Comment (ppenna) [inline tests/rust/pipe-test/create_pipe/mod.rs:34]: Would you add a full stop at the end of the comment line?

File: tests/rust/pipe-test/create_pipe/mod.rs [hunk 4]
```diff
@@ -19,6 +20,24 @@ use ::demikernel::{
...
+            let libos_name: LibOSName = LibOSName::from_env().expect("Should have a valid LibOS name").into();
+            LibOS::new(libos_name).expect("Should be able to create another libOS")
+        };
+
+        crate::collect!(result, crate::test!(create_pipe_with_name_and_close(libos, pipe_name)));
+    }
```
[the remark this answers is not in the document]
Comment (iyzhang) [reply tests/rust/pipe-test/create_pipe/mod.rs:40]: Yes, I am planning to move this to processes once we figure it out but I figured we would not do that for milestone 1.1

[the remark this answers is not in the document]
Comment (ppenna) [reply tests/rust/pipe-test/create_pipe/mod.rs:40]: https://github.com/demikernel/demikernel/issues/702

**Review by ppenna — APPROVED** [review]
LGTM

File: tests/rust/pipe-test/create_pipe/mod.rs (hunk 4 above)
Comment (ppenna) [inline tests/rust/pipe-test/create_pipe/mod.rs:40]: Please, wrap this in a standalone test function.

**Pull request merged**